packages/@expo/cli/src/start/platforms/android/AndroidSdk.ts

```diff
@@ -0,0 +1,37 @@
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+
+/**
+ * The default Android SDK locations per platform.
+ * @see https://developer.android.com/studio/run/emulator-commandline#filedir
+ * @see https://developer.android.com/studio/intro/studio-config#optimize-studio-windows
+ */
+const ANDROID_DEFAULT_LOCATION: Readonly<Partial<Record<NodeJS.Platform, string>>> = {
+  darwin: path.join(os.homedir(), 'Library', 'Android', 'sdk'), 
+  linux: path.join(os.homedir(), 'Android', 'sdk'),
+  win32: path.join(os.homedir(), 'AppData', 'Local', 'Android', 'Sdk'),
+};
+
+/**
+ * Resolve the root folder where the Android SDK has been installed.
+ * This checks both `ANDROID_HOME`, `ANDROID_SDK_ROOT`, and the default path for the current platform.
+ * @see https://developer.android.com/studio/command-line/variables
+ */
+export function resolveSdkRoot() {
+  if (process.env.ANDROID_HOME && fs.existsSync(process.env.ANDROID_HOME)) {
```

Comment:
We can rename this to assert, and try-catch it in the `ADBServer.ts`. Should have the same behavior, but with the exception stuff.

One thing to note is that users might have an old `ANDROID_HOME` or `ANDROID_SDK_ROOT` where one of them is invalid, and the other one is. (courtesy of Google not knowing what they want by going back and forth on the env names)

Reply:
Done, note that we don't fall back if `ANDROID_HOME` or `ANDROID_SDK_ROOT` anymore.

When one of them is invalid, it will warn with the assert message. In these cases, we will go directly to the globally installed `adb` executable.